packages/lit-dev-content/site/blog/2022-02-07-eleventy.md

```diff
@@ -0,0 +1,181 @@
+---
+tags: blogPosts
+layout: blog-post.html
+title: "Eleventy + Lit"
+summary: "Announcing a new Lit Labs Eleventy plugin for static pre-rendering of Lit components"
```

Comment:
Let's double check that we want to call this "static pre-rendering". I can't necessarily think of anything better, but I wonder if this is meaningful to everyone?

Reply:
Yes good point. I think matching the precise terminology used in https://developers.google.com/web/updates/2019/02/rendering-on-the-web could be a good idea.

Reply:
Reading it more carefully, I think this is a summary of the way the terms are defined in this article:

- *Static rendering*: Build-time + mostly works without JS
- *Prerendering*: Build-time + mostly does not work without JS
- *Server Rendering*: Rendered on the server + mostly works without JS
- *Client-Side Rendering (CSR)*: All rendering and data fetches done on the client
- *Universal Rendering* or *SSR*: Server Rendering + Client-Side Rendering with rehydration

One key snippet:

```
However, it’s important to understand the difference between static rendering and
prerendering: static rendered pages are interactive without the need to execute
much client-side JS, whereas prerendering improves the First Paint or First
Contentful Paint of a Single Page Application that must be booted on the client
in order for pages to be truly interactive.
```

It seems like by these definitions, we could be doing either "static rendering" or "prerendering" depending on the nature of the components.

I'm also a little confused that the article seems to distinguish between "Server Rendering" and "Server-Side Rendering"!

Reply:
The distinction between "Server Rendering" and "SSR" is one I've heard and used before. Basically it's calling out that we've always done some amount of "server side rendering" like with CGI and PHP. "Server rendering" is supposed to be distinct from SSR which is now deep rendering of previously client-side components on the server.

Static rendering vs prerendering is _not_ a distinction I've seen elsewhere.

My thoughts about "static pre-rendering" are:
* Do people understand "static"? Probbaly since this is for a static site generator?
* Do people understand "pre-rendering" as opposed to just "rendering"? I'm not sure I even do. 🤷‍♂️ All of eleventy is "pre" rendering, no?

So maybe "static rendering"?

Also, separately, I almost always see "hydration" not "rehydration"

Reply:
> So maybe "static rendering"?

Done. Makes sense, I like the alignment with "static site generator" in this context. Removed all use of "pre-render" in favor of "static rendering" or "statically rendered".


> Also, separately, I almost always see "hydration" not "rehydration"

Done.